scheduler/src/managers/node_agent/metrics.rs

```diff
@@ -35,14 +42,27 @@ pub struct NodeAgent {
 
 impl NodeAgent {
     /// Create a new `NodeAgent`.
-    pub fn new() -> Self {
+    pub fn new(id: String, address: &SocketAddr) -> Self {
         NodeAgent {
+            id,
+            address: *address,
             last_heartbeat: Local::now(),
             memory: None,
             cpu: None,
         }
     }
 
+    /// Get the Agent's id.
+    pub fn get_id(&self) -> String {
+        self.id.to_string()
+    }
+
+    /// Get the Agent's server address.
+    /// This is the URL that should be used to contact its gRPC server.
+    pub fn get_address(&self) -> String {
+        self.address.to_string()
+    }
```

Comment:
Remove the `get_` prefix. Return a borrowed string instead.
```suggestion
    pub fn address(&self) -> &str {
        self.address
    }
```